presto-spark-base/src/main/java/com/facebook/presto/spark/execution/LocalTask.java

```diff
@@ -0,0 +1,261 @@
+
+        ResponseHandler responseHandler = createAdaptingJsonResponseHandler(taskInfoCodec);
+        ListenableFuture<BaseResponse<TaskInfo>> future = httpClient.executeAsync(request, responseHandler);
+        FutureCallback callback = new SimpleHttpResponseHandler<>(new UpdateResponseHandler(), request.getUri(), stats.getHttpResponseStats(), LOCAL_TASK_ERROR);
+
+        Futures.addCallback(
+                future,
+                callback,
+                executor);
+    }
+
+    private synchronized void scheduleGetTaskInfo()
+    {
+        scheduledFuture = updateScheduledExecutor.scheduleWithFixedDelay(() -> {
+            try {
+                HttpUriBuilder uriBuilder = uriBuilderFrom(taskInfo.getTaskStatus().getSelf());
+                Request request = setContentTypeHeaders(false, prepareGet())
+                        .setUri(uriBuilder.build())
+                        .build();
+
+                ResponseHandler responseHandler = createAdaptingJsonResponseHandler(taskInfoCodec);
+                ListenableFuture<BaseResponse<TaskInfo>> future = httpClient.executeAsync(request, responseHandler);
+                FutureCallback callback = new SimpleHttpResponseHandler<>(new TaskInfoHandler(), request.getUri(), stats.getHttpResponseStats(), LOCAL_TASK_ERROR);
+
+                Futures.addCallback(
+                        future,
+                        callback,
+                        executor);
+            }
+            catch (Throwable t) {
+                throw t;
+            }
+        }, 0, 100, MILLISECONDS);
+    }
+
+    public synchronized void stop()
+    {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(true);
+        }
+    }
+
+    private class UpdateResponseHandler
+            implements SimpleHttpResponseCallback<TaskInfo>
+    {
+        @Override
+        public void success(TaskInfo value)
+        {
+            try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
+                log.error("success " + value.getTaskId());
+                taskInfo = value;
+            }
+        }
+
+        @Override
+        public void failed(Throwable cause)
+        {
+            try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
+                log.error("failed " + cause);
+            }
+        }
+
+        @Override
+        public void fatal(Throwable cause)
+        {
+            try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
+                log.error("fatal " + cause);
+            }
+        }
+    }
+
+    private class TaskInfoHandler
+            implements SimpleHttpResponseCallback<TaskInfo>
+    {
+        @Override
+        public void success(TaskInfo value)
+        {
+            try (SetThreadName ignored = new SetThreadName("TaskInfoHandler-%s", taskId)) {
+                log.error("TaskInfoHandler success " + value.getTaskId());
```

Comment:
Should it be log.info() ?

Reply:
or debug. This is a frequently used method. So info would create a lot of noise